contrib/internal/httputil/trace.go

```diff
@@ -54,6 +57,30 @@ func TraceAndServe(h http.Handler, cfg *TraceConfig) {
 
 	cfg.ResponseWriter = wrapResponseWriter(cfg.ResponseWriter, span)
 
+	op := httpinstr.StartHandlerOperation(
+		httpinstr.HandlerOperationArgs{
+			IsTLS:      cfg.Request.TLS != nil,
+			Method:     cfg.Request.Method,
+			Host:       cfg.Request.Host,
+			RequestURI: cfg.Request.RequestURI,
+			RemoteAddr: cfg.Request.RemoteAddr,
+			Headers:    cfg.Request.Header,
```

Comment:
Why not just pass the request?

Reply:
It is intentionally this way because:
1. We shouldn't trust the listeners. So we don't want to give them the pointer to the net/http request which shouldn't be kept for longer than the handler lifetime.
2. Pass the least amount of information. Which is all we need for the rule addresses + the intake event payload. This allows not to retain more memory than required.
3. The idea is to be able reuse this "abstract HTTP request" in every Go HTTP framework. I also like saying this is the contract interface between the HTTP integrations and the HTTP monitoring features (the WAF here). 

WDYT?